sda/cmd/rotatekey/rotatekey.go

```diff
@@ -0,0 +1,235 @@
+
+				continue
+			}
+
+			newHeader, err := reencrypt.CallReencryptHeader(header, pubKeyEncoded, conf.RotateKey.Grpc)
+			if err != nil {
+				msg := fmt.Sprintf("failed to rotate c4gh key for file %s", fileID)
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+			if newHeader == nil {
+				err := errors.New("reencrypt returned empty header")
+				msg := fmt.Sprintf("failed to rotate c4gh key for file %s", fileID)
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			// Rotate header in database
+			if err := db.StoreHeader(newHeader, fileID); err != nil {
+				msg := fmt.Sprintf("StoreHeader failed for file-id: %s", fileID)
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			// Rotate keyhash
+			if err := db.SetKeyHash(keyhash, fileID); err != nil {
+				msg := fmt.Sprintf("SetKeyHash failed for file-id: %s", fileID)
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			aID, err := db.GetAccessionID(fileID)
+			if err != nil {
+				msg := fmt.Sprintf("GetAccessionID failed for file-id: %s", fileID)
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			// Send re-verify message
+			reVerify, err := db.GetReVerificationData(aID)
+			if err != nil {
+				msg := fmt.Sprintf("GetReVerificationData failed for file-id %s", fileID)
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			reVerifyMsg, _ := json.Marshal(&reVerify)
+			err = schema.ValidateJSON(fmt.Sprintf("%s/ingestion-verification.json", conf.Broker.SchemasPath), reVerifyMsg)
+			if err != nil {
+				msg := "Validation of outgoing re-verify message failed"
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			if err := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "archived", reVerifyMsg); err != nil {
+				msg := "failed to publish message"
+				log.Errorf("%s, reason: %v", msg, err)
+				NackAndSendToErrorQueue(mq, delivered, conf.Broker.Exchange, msg, err.Error())
+
+				continue
+			}
+
+			if err := delivered.Ack(false); err != nil {
+				log.Errorf("failed to Ack message, reason: (%s)", err.Error())
+			}
+		}
+	}()
+
+	<-forever
+}
+
+// Nack message without requeue. Send the message to an error queue so it can be analyzed.
+func NackAndSendToErrorQueue(mq *broker.AMQPBroker, delivered amqp091.Delivery, mqExchange, msg, reason string) {
```

Comment:
```suggestion
// Ack message and send the payload to an error queue so it can be analyzed.
func AckAndSendToErrorQueue(mq *broker.AMQPBroker, delivered amqp091.Delivery, mqExchange, msg, reason string) {
```

Update the function name to reflect what's actually happening.